src/main.cpp

```diff
@@ -11,6 +11,8 @@
 #include <device.h>
 #include <drivers/gpio.h>
 
+#include <stm32_ll_system.h>
```

Comment:
This requires Zephyr 2.5 (generic LL headers patch) aswell, but master is still based on Zephyr 2.4, see [west.yml](https://github.com/LibreSolar/charge-controller-firmware/blob/master/zephyr/west.yml). For this reason, the continuous integration build (see the red X next to the commit hash) is also failing.

Reply:
I believe the current stable build from Zephyr is still 2.4.99.  Zephyr 2.5.xx is still in "working draft" AFAIK.

Also I don't build with west, but rather cmake: both are officially supported by Zephyr project. I haven't encountered any problem with cmake yet.

BTW, thanks for your efforts in this project!!  great job.!

Reply:
BTW, I am working with NiFe batteries, and chose your project because it was open and allowed the flexibility to tailor the system to handle the higher charging power used with NiFe batteries.   If you know of anyone who has worked with NiFe batteries in conjunction with your chargers, or has interest in them, please have them contact me.

Reply:
I think @xsider is working with NiFe batteries aswell according to [this forum post](https://talk.libre.solar/t/pwm2420-lus-for-battery-voltages-24v/175).

Regarding Zephyr version: The 2.4.99 is not stable, it's just the master branch between 2.4 release and 2.5 RC1, which has been cut last weekend, btw. However, I'm already trying out some features of the upcoming 2.5 release in the develop branch of this firmware.
We are maintaining an [own fork of Zephyr](https://github.com/LibreSolar/zephyr) which contains the stable branches + some cherry-picked updates which have not made it into upstream yet (e.g. because they are still in a PR). So if you are building with plain cmake (which is totally fine, of course), you should still pull the forked branch specified in the `west.yml` to be sure that it works well.

Reply:
Thats correct. I work with NiFe (400Ah) cells, beside to Pb (wet and gel), LiFeYPo, LiPo on different chargers, either commercial or libresolar. Except the incident mentioned above, i had no trouble so far with NiFe.